app/lib/methods/helpers/buildImageURL.ts

```diff
@@ -0,0 +1,20 @@
+export const isValidLink = (link: string): boolean => {
```

Comment:
I can see you reused from backend, but can you move to its own file?
You're exporting, so it might be useful to have it outside image's context.

Can you also rename it to `isValidUrl`?
It makes use of `URL` api, so calling it url instead of link makes more sense.